clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -1192,13 +1192,22 @@ raw2trace_t::process_syscall_pt(raw2trace_thread_data_t *tdata, uint64_t syscall
         tdata->pt_decode_state_ = std::unique_ptr<drir_t>(new drir_t(GLOBAL_DCONTEXT));
     }
     tdata->pt_decode_state_->clear_ilist();
+    uint64_t syscall_decode_recoverable_error_count = 0;
     pt2ir_convert_status_t pt2ir_convert_status = tdata->pt2ir.convert(
-        pt_data->data.get(), pt_data_size, tdata->pt_decode_state_.get());
+        pt_data->data.get(), pt_data_size, tdata->pt_decode_state_.get(),
+        &syscall_decode_recoverable_error_count);
     if (pt2ir_convert_status != PT2IR_CONV_SUCCESS) {
-        tdata->error = "Failed to convert PT raw trace to DR IR [error status: " +
-            std::to_string(pt2ir_convert_status) + "]";
-        return false;
+        if (!pt2ir_allow_recoverable_errors_) {
```

Comment:
Aren't there other types of errors?  This assumes failure only happens with "recoverable errors".  But the pt2ir code had a limit on retries so won't it fail sometimes yet the code here will never consider it to fail?

Reply:
Added a conjunction for the pt2ir error type for "next instr decoding".